File: LinkedList/04/sol.java
// { Driver Code Starts
import java.util.*;
import java.lang.*;

class Node
{
    int data;
    Node next;
    Node(int key)
    {
        data = key;
        next = null;
    }
}

class ReverseInSize
{
    static Node head;
    
    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int a1 = sc.nextInt();
            Node head = new Node(a1);
            Node tail = head;
            for(int i = 1; i < n; i++)
            {
                int a = sc.nextInt();
                // addToTheLast(new Node(a));
                tail.next = new Node(a);
                tail =tail.next;
            }
            
            int k = sc.nextInt();
            GfG gfg = new GfG();
            Node res = gfg.reverse(head, k);
            printList(res);
            System.out.println();
        }
    }
    
    public static void printList(Node node)
    {
        while(node != null)
        {
            System.out.print(node.data + " ");
            node = node.next;
        }
    }
    
}

// } Driver Code Ends


/*node class of the linked list

class Node
{
    int data;
    Node next;
    Node(int key)
    {
        data = key;
        next = null;
    }
}

*/

class GfG
{
    public static Node reverse(Node node, int k)
    {
        boolean flag = true;
        
        Node head = node;
        Node lastNode = null;
        Node temp = null;
        Node ptr = node;
        Node frontHead = null;
        while(true)
        {
            lastNode = head;
            ptr = head.next;
            for (int i = 1; i < k; i++)
            {
                if(ptr == null)break;
                temp = ptr.next;
                ptr.next = head;
                head = ptr;
                ptr = temp;
            }
            if(ptr == null)
            {
                lastNode.next = null;
                if(frontHead == null)
                {
                    node = head;
                    return node;
                }
                frontHead.next = head;
                break;
            }
            if(frontHead== null)
            {
                node = head;
                frontHead = lastNode;
            }
            else
            {
                frontHead.next = head;
                frontHead = lastNode;
            }
            head = ptr;
        }
        return node;
        //Your code here
    }
}

File: LinkedList/13/sol.py
#User function Template for python3

''' Node for linked list:

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

'''
def reverse(head):
    ptr = head.next
    lastNode = head
    while ptr is not None:
        temp = ptr.next
        ptr.next = head
        head = ptr
        ptr = temp
    lastNode.next = None
    return head
    

def addLists(first, second):
    #printList(first)
    #printList(second)
    first = reverse(first)
    second = reverse(second)
    #printList(first)
    #printList(second)
    carry = 0
    head = first
    while first.next is not None and second.next is not None:
        first.data += second.data + carry
        if first.data > 9:
            first.data %= 10
            carry = 1
        else:
            carry = 0
        first = first.next
        second = second.next
    first.data += second.data + carry
    if first.data > 9:
        first.data %= 10
        carry = 1
    else:
        carry = 0
    if first.next is None:
        first.next = second.next
    while carry == 1:
        if first.next is None:
            first.next = Node(1)
            carry = 0
        else:
            first.next.data += 1
            if first.next.data > 9:
                first.next.data %= 10
                first = first.next
                carry = 1
            else:
                carry = 0
                
    return reverse(head)
                
                
    # code here
    # return head of sum list



#{ 
#  Driver Code Starts
#Initial Template for Python 3

# Node Class
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

# Linked List Class
class LinkedList:
    def __init__(self):
        self.head = None
        self.tail = None

    # creates a new node with given value and appends it at the end of the linked list
    def insert(self, val):
        if self.head is None:
            self.head = Node(val)
            self.tail = self.head
        else:
            self.tail.next = Node(val)
            self.tail = self.tail.next

# prints the elements of linked list starting with head
def printList(n):
    while n:
        print(n.data,end=' ')
        n = n.next
    print()

if __name__ == '__main__':
    for _ in range(int(input())):
        
        n = int(input())
        arr1 = ( int(x) for x in input().split() )
        LL1 = LinkedList()
        for i in arr1:
            LL1.insert(i)
        
        m = int(input())
        arr2 = ( int(x) for x in input().split() )
        LL2 = LinkedList()
        for i in arr2:
            LL2.insert(i)
        
        res = addLists(LL1.head, LL2.head)
        printList(res)
# } Driver Code Ends